### ISSUE TITLE
Testing Revit Family type for equality against selected Family in project fails

### ISSUE THREAD

**Issue by ThomasMahon** [DynamoDS/DynamoRevit#997]
## Dynamo version

1.0.0
## Revit version

Revit 2016
## Operating system

Win 10
## What did you do?

I have to test an array of adaptive component Family panels for equality against a particular type. The test is required as there are a number of different types of these components in the project: solid, window, etc and only the window components need to be identified/selected as they contain parameters which require updating (window opening width/height) from data stored in an Excel spreadsheet.

I created an imperative DesignScript definition which generates the test Family Type using the Family.Types query method and tests this type for equality against the family type of the selected components from the project.
## What did you expect to see?

I expected that the equality test would return true when the test Family type and the extracted Family type from the component match (to then be able to update its parameters from the Excel data)
## What did you see instead?

The equality test fails, despite even the object IDs being identical.

And if the objects IDs are used for equality, it works!?

The following example illustrates the behaviour I experienced in the DesignScript definition:

![revit family eqality fails](https://cloud.githubusercontent.com/assets/18601851/15341096/990f87f6-1c84-11e6-807c-6b7ca19eee00.png)

**Comment by ksobon** [comment]
I am not seeing that same behavior. 

![image](https://cloud.githubusercontent.com/assets/529781/26288335/c8358a02-3e5d-11e7-85ab-0d7c0481b8eb.png)

Is this still an issue in 1.3?

**Comment by ThomasMahon** [comment]
It's still an issue in 1.3 if this is anything to go by:

![image](https://cloud.githubusercontent.com/assets/18601851/26326941/6279f4f4-3f35-11e7-83f5-ad6efa50eb5d.png)

**Comment by ThomasMahon** [comment]
I just noticed you are using `Object.Equals` instead of `==` (from the Operators library), which does work. Two nodes to do the same thing seems odd. In any case, I would expect both to give the same result since either node accepts any type.

**Comment by ksobon** [comment]
yes, and no. it's good coding practice to override the `==` operator if one overrides the `Equals`. It seems like that is not the case with Dynamo and particularly this `ElementType` object. I can fix that. Thanks for catching. cc: @mjkkirschner @jnealb

**Comment by ksobon** [comment]
@ThomasMahon actually that I have looked at this there is nothing wrong with the current implementation: like I said before `==` and `Equals` are two different methods that check two different things. 

Official MSDN documentation: 

_In C#, there are two different kinds of equality: reference equality and value equality. Value equality is the commonly understood meaning of equality: it means that two objects contain the same values. For example, two integers with the value of 2 have value equality. Reference equality means that there are not two objects to compare. Instead, there are two object references, both of which refer to the same object._

Basically it means that you can do this:
![image](https://cloud.githubusercontent.com/assets/529781/26336134/cef94766-3f3e-11e7-8b9c-277818216a47.png)

It's all working as expected. When the same object is referenced the `==` operator returns true while `Equals` returns true even if two different objects are used that have the same values. 

I think you just need to use the "right" node from now on.

**Comment by ThomasMahon** [comment]
That does figure. I wasn't aware of the `Equals` node and assumed `==` node could be used regardless of value or reference.
